deps/rabbit/src/rabbit_variable_queue.erl

```diff
@@ -2435,3 +2436,15 @@ maybe_client_terminate(MSCStateP) ->
         _:_ ->
             ok
     end.
+
+format_state(#vqstate{ram_pending_ack = RamPendingAck,
+                      disk_pending_ack = DiskPendingAck} = S) ->
+    S#vqstate{ram_pending_ack = format_state_map_element(ram_pending_ack, RamPendingAck),
+              disk_pending_ack = format_state_map_element(disk_pending_ack, DiskPendingAck),
+              index_state = index_state_truncated,
+              store_state = store_state_truncated}.
```

Comment:
This means we can't debug CQ issues at all anymore. I understand removing the message contents, but this is probably too much.

Reply:
I was wondering about this. I'll add a configuration setting to clean up state so that it's configurable, and have cleanup disabled by default.

In the case of index state `qi` record, I noticed that `confirms` could have a large number of entries, so I'll "summarize" that set like I do with other items.

Looking at `store_state`, it looks like the `write_buffer` and `cache` could potentially have a lot of data to be summarized.

Do you have any other suggestions to reduce the size of the logged state without losing important information?

Reply:
Changes implemented, thanks @lhoguin